### PR TITLE
ARROW-2756: [Python] Remove redundant imports and minor fixes in parquet tests

### PR DESCRIPTION
**Pull request by kszucs**

### CODE REVIEW

File: python/pyarrow/tests/test_parquet.py [hunk 1]
```diff
@@ -69,10 +66,7 @@ def _check_roundtrip(table, expected=None, **params):
         expected = table
 
     result = _roundtrip_table(table, **params)
-    if not result.equals(expected):
-        print(expected)
-        print(result)
-        assert result.equals(expected)
+    assert result.equals(expected)
 
     result = _roundtrip_table(result, **params)
```
Comment (kszucs) [inline python/pyarrow/tests/test_parquet.py]: Does this intentionally check the roundtrip twice?

Comment (wesm) [reply python/pyarrow/tests/test_parquet.py]: Looking at where it was added https://github.com/apache/arrow/commit/3b69c5acafaccb211f232b586a4e0b1172809809, I think you can remove the redundant check

Comment (kszucs) [reply python/pyarrow/tests/test_parquet.py]: done

Comment (pitrou) [reply python/pyarrow/tests/test_parquet.py]: My memory is fuzzy, but I _think_ the point was to test twice.

Comment (wesm) [reply python/pyarrow/tests/test_parquet.py]: @pitrou you're right, I misread the roundtrip check, which is different from the previous one. @kszucs can you restore those lines and rebase?

Comment (kszucs) [reply python/pyarrow/tests/test_parquet.py]: Sure, will do

File: python/pyarrow/tests/test_parquet.py [hunk 2]
```diff
@@ -29,21 +29,19 @@
...
 import numpy as np
 import pandas as pd
-
+import pyarrow as pa
+import pyarrow.parquet as pq
 import pandas.util.testing as tm
 
 # Ignore these with pytest ... -m 'not parquet'
-parquet = pytest.mark.parquet
+pytestmark = pytest.mark.parquet
```
Comment (pitrou) [inline python/pyarrow/tests/test_parquet.py]: A nit, but "pytestmark" is informative. Call it "parquet_test" perhaps?

File: python/pyarrow/tests/test_parquet.py [hunk 3]
```diff
@@ -29,21 +29,19 @@
...
 from pyarrow.tests import util
 from pyarrow.filesystem import LocalFileSystem
-import pyarrow as pa
 from .pandas_examples import dataframe_with_arrays, dataframe_with_lists
 
 import numpy as np
 import pandas as pd
-
+import pyarrow as pa
+import pyarrow.parquet as pq
```
Comment (pitrou) [inline python/pyarrow/tests/test_parquet.py]: Uh, this test file should be skipped if parquet support wasn't built, instead of raising.

File: python/pyarrow/tests/test_parquet.py [hunk 4]
```diff
@@ -89,22 +83,21 @@ def _roundtrip_pandas_dataframe(df, write_kwargs):
     return table1.to_pandas()
 
 
-@parquet
```
Comment (pitrou) [inline python/pyarrow/tests/test_parquet.py]: Why did you remove the decorators instead of switching to the new name?

**Review by pitrou — CHANGES_REQUESTED** [review]

File: python/pyarrow/tests/test_parquet.py (hunk 2 above)
Comment (kszucs) [reply python/pyarrow/tests/test_parquet.py]: This is a module level marker, see the end of [this section](https://docs.pytest.org/en/latest/example/markers.html#marking-whole-classes-or-modules)

File: python/pyarrow/tests/test_parquet.py (hunk 3 above)
Comment (kszucs) [reply python/pyarrow/tests/test_parquet.py]: Sure, we can use `pytest.importorskip` for that

File: python/pyarrow/tests/test_parquet.py (hunk 2 above)
Comment (pitrou) [reply python/pyarrow/tests/test_parquet.py]: Gasp. Can you add a comment?

File: python/pyarrow/tests/test_parquet.py (hunk 4 above)
Comment (kszucs) [reply python/pyarrow/tests/test_parquet.py]: Because I've switched to a module level marker instead

File: python/pyarrow/tests/test_parquet.py (hunk 2 above)
Comment (kszucs) [reply python/pyarrow/tests/test_parquet.py]: Sure

File: python/pyarrow/tests/test_parquet.py (hunk 3 above)
Comment (wesm) [reply python/pyarrow/tests/test_parquet.py]: Be careful with `importorskip` because it can silently ignore errors due to linking issues  library loading. These should raise

Comment (wesm) [reply python/pyarrow/tests/test_parquet.py]: BTW the issue with `importorskip` is exactly why the imports are inside the tests

Comment (kszucs) [reply python/pyarrow/tests/test_parquet.py]: Thanks for the warning! Do we really want to skip if parquet is not available? It wasn't skipped previously.

Comment (pitrou) [reply python/pyarrow/tests/test_parquet.py]: Hmm, I think it was skipped. The tests wouldn't fail if parquet wasn't built.

Comment (kszucs) [reply python/pyarrow/tests/test_parquet.py]: Don't the tests fail in both cases?

Comment (wesm) [reply python/pyarrow/tests/test_parquet.py]: Unless something changed recently, the tests in this file were only run if either `--parquet` was passed or `import pyarrow.parquet as pq` runs successfully

see

https://github.com/apache/arrow/blob/master/python/pyarrow/tests/conftest.py#L39

Comment (kszucs) [reply python/pyarrow/tests/test_parquet.py]: Fall back to import `pq` in each test function, I'll submit another PR to handle optional dependencies without redundant imports after 0.10.0 release.

**Review by wesm — APPROVED** [review]
+1. It's good to know about the module-level marker!

I checked that the skipping behavior is as intended:

broken extension:

```
(arrow-dev) 17:02 ~/code/arrow/python  (cleanup_parquet_tests)$ py.test pyarrow/tests/test_parquet.py 
=========================================== test session starts ============================================
platform linux -- Python 3.6.5, pytest-3.6.0, py-1.5.3, pluggy-0.6.0
rootdir: /home/wesm/code/arrow/python, inifile: setup.cfg
collected 86 items                                                                                         

pyarrow/tests/test_parquet.py ssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss [ 81%]
ssssssssssssssss                                                                                     [100%]

======================================== 86 skipped in 0.08 seconds ========================================
```

But we get failures with a broken extension if we pass `--parquet`

```
(arrow-dev) 17:02 ~/code/arrow/python  (cleanup_parquet_tests)$ py.test pyarrow/tests/test_parquet.py --parquet
=========================================== test session starts ============================================
platform linux -- Python 3.6.5, pytest-3.6.0, py-1.5.3, pluggy-0.6.0
rootdir: /home/wesm/code/arrow/python, inifile: setup.cfg
collected 86 items                                                                                         

pyarrow/tests/test_parquet.py FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFsFFFFFFFFFFF [ 81%]
FFFFFsFFFFFFFxFF                                                                                     [100%]
```
